homeassistant/components/zwave_js/config_flow.py

```diff
@@ -1498,6 +1497,9 @@ async def async_step_esphome(
             self.hass.config_entries.async_schedule_reload(existing_entry.entry_id)
             return self.async_abort(reason="already_configured")
 
+        # We are not aborting if home ID configured here, we just want to make sure that it's set
+        # We will update a USB based config entry automatically in `async_step_finish_addon_setup_user`
+        await self.async_set_unique_id(str(discovery_info.zwave_home_id))
```

Comment:
What was the reason for not setting unique id before this change?

Reply:
Setting a unique id isn't only about aborting if the same unique id is already configured, but also to abort if the same unique id is in progress.